faunadb-scala/src/main/scala/faunadb/values/Codec.scala

```diff
@@ -131,11 +169,26 @@ object Decoder {
       }
   }
 
-  class SeqDecoder[T: Decoder] extends Decoder[Seq[T]] {
+  class ArrayDecoder[T: Decoder](implicit ev: ClassTag[T]) extends Decoder[Array[T]] {
+    def decode(v: Value, path: FieldPath): Result[Array[T]] =
+      Field.collect(Field.to[T]).get(v).map(x => x.toArray[T])
+  }
+
+  implicit def GenericArrayDecoder[T: Decoder](implicit ev: ClassTag[T]) =
```

Comment:
Is it necessary? It's resolving the decoder correctly, so if you could provide an example would be nice